src/neo/Plugins/Plugin.cs

```diff
@@ -58,74 +59,112 @@ protected Plugin()
             Configure();
         }
 
-        public abstract void Configure();
-
-        protected virtual void OnPluginsLoaded()
+        protected virtual void Configure()
         {
         }
 
         private static void ConfigWatcher_Changed(object sender, FileSystemEventArgs e)
         {
-            foreach (var plugin in Plugins)
+            switch (GetExtension(e.Name))
             {
-                if (plugin.ConfigFile == e.FullPath)
-                {
-                    plugin.Configure();
-                    plugin.Log($"Reloaded config for {plugin.Name}");
+                case ".json":
+                    Plugins.FirstOrDefault(p => p.ConfigFile == e.FullPath)?.Configure();
                     break;
-                }
+                case ".dll":
+                    if (e.ChangeType != WatcherChangeTypes.Created) return;
+                    if (GetDirectoryName(e.FullPath) != PluginsDirectory) return;
+                    try
+                    {
+                        LoadPlugin(Assembly.Load(File.ReadAllBytes(e.FullPath)));
+                    }
+                    catch { }
+                    break;
+            }
+        }
+
+        private static Assembly CurrentDomain_AssemblyResolve(object sender, ResolveEventArgs args)
+        {
+            if (args.Name.Contains(".resources"))
+                return null;
+
+            Assembly assembly = AppDomain.CurrentDomain.GetAssemblies().FirstOrDefault(a => a.FullName == args.Name);
+            if (assembly != null)
+                return assembly;
+
+            AssemblyName an = new AssemblyName(args.Name);
+            string filename = an.Name + ".dll";
+
+            try
+            {
+                return Assembly.LoadFrom(filename);
```

Comment:
```suggestion
                return Assembly.Load(File.ReadAllBytes(filename));
```

In order to free the file if you want to uninstall it later

Reply:
What do you mean @shargon ?